### PR TITLE
Vectorize solarization.

### LINKED ISSUE #1463: Vectorize Solarization.
**Short Description**
`Solarization` can be easily ported to use `VectorizedBaseImageAugmentationLayer`.

The implementation basically only changes:
`addition_factor(dtype=self.compute_dtype)` to `addition_factor(shape=(batch_size, 1, 1, 1), ...)`

**Performance summary**
* Eager -> improvement.
* Graph -> similar.
* XLA -> similar.

Graphs:
comparison
comparison_no_old_eager

If you approve I can submit a PR.

### PR DESCRIPTION
**Pull request by sebastian-sz**
# What does this PR do?
Modifies `Solarization` to use vectorized approach.

Fixes #1463 

The performance is similar in graph / XLA mode and better in eager mode. 
If random behavior is desired, different augmentations are applied per image.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/keras-team/keras-cv/blob/master/.github/CONTRIBUTING.md),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue? Please add a link
      to it if that's the case.
- [x] Did you write any new necessary tests?
- [ ] If this adds a new model, can you run a few training steps on TPU in Colab to ensure that no XLA incompatible OP are used?

## Who can review?
@LukeWood @ianstenbit

### CODE REVIEW

File: keras_cv/layers/preprocessing/solarization.py [hunk 1]
```diff
@@ -93,41 +93,51 @@ def __init__(
         )
         self.value_range = value_range
 
-    def get_random_transformation(self, **kwargs):
+    def get_random_transformation_batch(self, batch_size, **kwargs):
         return (
-            self.addition_factor(dtype=self.compute_dtype),
-            self.threshold_factor(dtype=self.compute_dtype),
+            self.addition_factor(
```
Comment (LukeWood) [inline keras_cv/layers/preprocessing/solarization.py:98]: pretty tiny nit, but can we change the random transformation format from:

```
(addition, threshold)
```

to

```
{
"addition": addition,
"threshold": threshold
}
```
?

The reason I think this is better is that we are seeing some usage of the get_transformation() APIs directly.

Comment (sebastian-sz) [reply keras_cv/layers/preprocessing/solarization.py]: @LukeWood done. Changed `get_random_transformation_batch` to return a dictionary.

**Review by LukeWood — APPROVED** [review]
LGTM, let me do one more pass then will merge.  Thanks as always @sebastian-sz

**Comment by sebastian-sz** [comment]
@LukeWood thanks! No need to rush, I know this is low prio.

**Review by LukeWood — APPROVED** [review]

**Comment by LukeWood** [comment]
Thanks @sebastian-sz !  Always appreciate the quality PRs!

**Pull request merged**